ethereum/watcher/watcher.go

```diff
@@ -0,0 +1,100 @@
+package watcher
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	eth "github.com/ethereum/go-ethereum"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+var checkForBlocksTimeout = time.Second
+
+// EventFilterer filters the chain for specific events (logs).
```

Comment:
Is this a typo with the intended name being `EventFilter` (without the extra `er`)?

Reply:
not really, I guess I named it this since there was a `FilterLogs` function which this calls, making it a `Filterer` in a way, lol. I think either one is fine, can change if this bothers you?

Reply:
I think it is just such an awkward word to say, that people avoid it. It's like calling someone that answers questions for a living an answerer. Whatever you decide is fine. I just wanted to make sure it wasn't a typo. :)

https://github.com/search?q=eventFilter&type=code (1 million results)
https://github.com/search?q=eventFilterer&type=code (28 results)